lfs/hook.go

```diff
@@ -27,17 +28,30 @@ func (h *Hook) Exists() bool {
 }
 
 // Path returns the desired (or actual, if installed) location where this hook
-// should be installed, relative to the local Git directory.
+// should be installed. It returns an absolute path in all cases.
 func (h *Hook) Path() string {
-	return filepath.Join(config.LocalGitDir, "hooks", string(h.Type))
+	return filepath.Join(h.Dir(), h.Type)
+}
+
+// Dir returns the directory used by LFS for storing Git hooks. By default, it
+// will return the hooks/ sub-directory of the local repository's .git
+// directory. If `core.hooksPath` is configured and supported (Git verison is
+// greater than "2.9.0"), it will return that instead.
+func (h *Hook) Dir() string {
+	customHooksSupported := git.Config.IsGitVersionAtLeast("2.9.0")
+	if hp, ok := config.Config.GitConfig("core.hooksPath"); ok && customHooksSupported {
+		return hp
+	}
```

Comment:
I think I may keep this as-is because then we'd have to do the `filepath.Join` up front, regardless of whether or not `core.hooksPath` is supported. I have a feeling that a call to `filepath.Join` is not as efficient as a map lookup, but this is such an uncommon operation I think either way would be fine 😄